src/os/structs/nodeutils.js

```diff
@@ -0,0 +1,21 @@
+goog.provide('os.structs.nodeutils');
+
+
+/**
+ * Flattens a tree into a specified array - optionally filtering.
+ * @param {os.structs.ITreeNode} node
+ * @param {Array<os.structs.ITreeNode>} results
+ * @param {function(os.structs.ITreeNode): boolean} opt_filter
+ */
+os.structs.nodeutils.flattenTree = function(node, results, opt_filter) {
+  if (opt_filter ? opt_filter(node) : true) {
```

Comment:
One alternative here is to use a default function:
```
opt_filter = opt_filter || goog.functions.TRUE;
if (opt_filter(node)) {
  ...
```

Reply:
I went with your suggestion - good tip.
Thanks for catching all these things!